### ISSUE TITLE
Questions for torchsparse.nn.BatchNorm module

### ISSUE THREAD

**Issue by louzq16** [mit-han-lab/torchsparse#97]
Thank for your great open-source torchsparse library!
I met some problems when using the torchsparse.nn.BatchNorm module.  
As shown in the figure,  occasionally the feature shape of sparse tensor is [1,C] during training, which causing the ValueError of torch.nn.BatchNorm.
![issue_figure](https://user-images.githubusercontent.com/44044820/125794110-415f72c2-6c0d-4a78-866a-922473af71ba.png)

Now  I try to maintain an addional mirror torchsparse.nn.BatchNorm module in eval mode, which is used to process the situation above.  I don't know if that method is correct. Would you please give a better approach to handle such problem?
![active_BN](https://user-images.githubusercontent.com/44044820/125795635-ebb43250-6bf4-41ab-81d4-614f46af46d9.png)

**Comment by CCInc** [comment]
Hi,

Our batchnorm is simply a wrapper around `torch.nn.BatchNorm1d`, so it is bound by the same constraints. Are you asking for training or inference? For inference you need to set `model.eval()` so that the batchnorm layer gets deactivated.

**Comment by NoOneUST** [comment]
> Hi,
> 
> Our batchnorm is simply a wrapper around `torch.nn.BatchNorm1d`, so it is bound by the same constraints. Are you asking for training or inference? For inference you need to set `model.eval()` so that the batchnorm layer gets deactivated.

I met the same problem. It happens while training.

**Comment by zhijian-liu** [comment]
As @CCInc mentioned, this is the expected behavior of batch normalization, which requires a batch size larger than 1 during training (in order to compute the batch mean and variance for normalization). You could try to use a larger batch size to resolve this issue.

**Comment by CCInc** [comment]
If there's any update on this issue feel free to reopen.
